[Bug]:[Nightly]Search failed reporting 'Timestamp lag too large lag(464829h12m59.941s) max(24h0m0s) err'

## Environment
- Milvus version:2.2.0(e68374b)
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):   kafka 
- SDK version(e.g. pymilvus v2.0.0rc2):2.2.1.dev5
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: 

## Current Behavior:
Search failed reporting 'Timestamp lag too large lag(464829h12m59.941s) max(24h0m0s) err'.
```
[2023-01-10T21:25:38.471Z] ___ TestCollectionSearch.test_search_multi_collections[5000-2-128-True-True] ___
[2023-01-10T21:25:38.471Z] [gw3] linux -- Python 3.8.16 /usr/local/bin/python3
[2023-01-10T21:25:38.471Z] 
[2023-01-10T21:25:38.471Z] self = <test_search.TestCollectionSearch object at 0x7f82a58d8f10>, nb = 5000
[2023-01-10T21:25:38.471Z] nq = 2, dim = 128, auto_id = True, _async = True
[2023-01-10T21:25:38.471Z] 
[2023-01-10T21:25:38.471Z]     @pytest.mark.tags(CaseLabel.L2)
[2023-01-10T21:25:38.471Z]     def test_search_multi_collections(self, nb, nq, dim, auto_id, _async):
[2023-01-10T21:25:38.471Z]         """
[2023-01-10T21:25:38.471Z]         target: test search multi collections of L2
[2023-01-10T21:25:38.471Z]         method: add vectors into 10 collections, and search
[2023-01-10T21:25:38.471Z]         expected: search status ok, the length of result
[2023-01-10T21:25:38.471Z]         """
[2023-01-10T21:25:38.471Z]         self._connect()
[2023-01-10T21:25:38.471Z]         collection_num = 10
[2023-01-10T21:25:38.471Z]         for i in range(collection_num):
[2023-01-10T21:25:38.471Z]             # 1. initialize with data
[2023-01-10T21:25:38.471Z]             log.info("test_search_multi_collections: search round %d" % (i + 1))
[2023-01-10T21:25:38.471Z]             collection_w, _, _, insert_ids = self.init_collection_general(prefix, True, nb,
[2023-01-10T21:25:38.471Z]                                                                           auto_id=auto_id,
[2023-01-10T21:25:38.471Z]                                                                           dim=dim)[0:4]
[2023-01-10T21:25:38.471Z]             # 2. search
[2023-01-10T21:25:38.471Z]             vectors = [[random.random() for _ in range(dim)] for _ in range(nq)]
[2023-01-10T21:25:38.471Z]             log.info("test_search_multi_collections: searching %s entities (nq = %s) from collection %s" %
[2023-01-10T21:25:38.471Z]                      (default_limit, nq, collection_w.name))
[2023-01-10T21:25:38.471Z] >           collection_w.search(vectors[:nq], default_search_field,
[2023-01-10T21:25:38.471Z]                                 default_search_params, default_limit,
[2023-01-10T21:25:38.471Z]                                 default_search_exp, _async=_async,
[2023-01-10T21:25:38.471Z]                                 check_task=CheckTasks.check_search_results,
[2023-01-10T21:25:38.471Z]                                 check_items={"nq": nq,
[2023-01-10T21:25:38.471Z]                                              "ids": insert_ids,
[2023-01-10T21:25:38.471Z]                                              "limit": default_limit,
[2023-01-10T21:25:38.472Z]                                              "_async": _async})
[2023-01-10T21:25:38.472Z] 
[2023-01-10T21:25:38.472Z] testcases/test_search.py:2458: 
[2023-01-10T21:25:38.472Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2023-01-10T21:25:38.472Z] utils/wrapper.py:33: in inner_wrapper
[2023-01-10T21:25:38.472Z]     res, result = func(*args, **kwargs)
[2023-01-10T21:25:38.472Z] base/collection_wrapper.py:164: in search
[2023-01-10T21:25:38.472Z]     check_result = ResponseChecker(res, func_name, check_task, check_items, check,
[2023-01-10T21:25:38.472Z] check/func_check.py:57: in run
[2023-01-10T21:25:38.472Z]     result = self.check_search_results(self.response, self.func_name, self.check_items)
[2023-01-10T21:25:38.472Z] check/func_check.py:216: in check_search_results
[2023-01-10T21:25:38.472Z]     search_res = search_res.result()
[2023-01-10T21:25:38.472Z] /usr/local/lib/python3.8/site-packages/pymilvus/orm/future.py:30: in result
[2023-01-10T21:25:38.472Z]     return self.on_response(self._f.result())
[2023-01-10T21:25:38.472Z] /usr/local/lib/python3.8/site-packages/pymilvus/client/asynch.py:181: in result
[2023-01-10T21:25:38.472Z]     self.exception()
[2023-01-10T21:25:38.472Z] /usr/local/lib/python3.8/site-packages/pymilvus/client/asynch.py:237: in exception
[2023-01-10T21:25:38.472Z]     raise self._exception
[2023-01-10T21:25:38.472Z] /usr/local/lib/python3.8/site-packages/pymilvus/client/asynch.py:227: in done
[2023-01-10T21:25:38.472Z]     self._results = self.on_response(self._response)
[2023-01-10T21:25:38.472Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2023-01-10T21:25:38.472Z] 
[2023-01-10T21:25:38.472Z] self = <pymilvus.client.asynch.ChunkedSearchFuture object at 0x7f82892397c0>
[2023-01-10T21:25:38.472Z] response = [status {
[2023-01-10T21:25:38.472Z]   error_code: UnexpectedError
[2023-01-10T21:25:38.472Z]   reason: "fail to search on all shard leaders, err=fail to Search, QueryNode ID=5, reason=Search 5 failed, reason Timestamp lag too large lag(464829h12m59.941s) max(24h0m0s) err %!w(<nil>)"
[2023-01-10T21:25:38.472Z] }
[2023-01-10T21:25:38.472Z] ]
[2023-01-10T21:25:38.472Z] 
[2023-01-10T21:25:38.472Z]     def on_response(self, response):
[2023-01-10T21:25:38.472Z]         for raw in response:
[2023-01-10T21:25:38.472Z]             if raw.status.error_code != 0:
[2023-01-10T21:25:38.472Z] >               raise MilvusException(raw.status.error_code, raw.status.reason)
[2023-01-10T21:25:38.472Z] E               pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=5, reason=Search 5 failed, reason Timestamp lag too large lag(464829h12m59.941s) max(24h0m0s) err %!w(<nil>))>
[2023-01-10T21:25:38.472Z] 
[2023-01-10T21:25:38.472Z] /usr/local/lib/python3.8/site-packages/pymilvus/client/asynch.py:245: MilvusException
```

## Expected Behavior:
success

## Report
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/2.2.0/60/pipeline/144

details:
1. log: [artifacts-milvus-distributed-kafka-nightly-60-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/2.2.0/60/artifact/artifacts-milvus-distributed-kafka-nightly-60-pymilvus-e2e-logs.tar.gz)
2. collection name: search_collection_fuGZyKAg
3. failed time: [2023-01-10T21:13:04.997Z] [gw3] [ 99%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_multi_collections[5000-2-128-True-True]

--- a/internal/querynode/task_read.go
+++ b/internal/querynode/task_read.go
@@ -170,7 +170,7 @@ func (b *baseReadTask) Ready() (bool, error) {
 	if guaranteeTs > serviceTime {
 		lag := gt.Sub(st)
 		maxLag := Params.QueryNodeCfg.MaxTimestampLag.GetAsDuration(time.Second)
-		if lag > maxLag {
+		if serviceTime != 0 && lag > maxLag {
 			log.Warn("guarantee and servicable ts larger than MaxLag",
 				zap.Time("guaranteeTime", gt),
 				zap.Time("serviceableTime", st),

